src/virt.c

```diff
@@ -578,15 +614,26 @@ lv_read (void)
                     &stats, sizeof stats) != 0)
             continue;
 
+	char *type_instance = NULL;
+	if( blockdevice_format_basename && blockdevice_format == source ) //valid only if we use "source" (full path to the device)
+	{
```

Comment:
Please remove curly braces around single-statement blocks, i.e.:

```c
if (cond)
  type_instance = "…";
else
  type_instance = "…";
```

Same below for `bd_xmlpath`.